source/LinkedList.inl

```diff
@@ -263,3 +312,33 @@ DataStructures::LinkedList<T>::map(std::function<T(const T&)> closure)
     current_element = current_element->next;
   }
 }
+
+template<typename T>
+bool
+operator==(LinkedList<T> a, LinkedList<T> b)
```

Comment:
These all want to take a const reference rather than a value. Otherwise, you're copying the entire list each time you use one of these operators.